context/app/static/js/components/bulkDownload/hooks.ts

```diff
@@ -0,0 +1,199 @@
+import { useCallback, useEffect } from 'react';
+import { useForm } from 'react-hook-form';
+import { zodResolver } from '@hookform/resolvers/zod';
+import { z } from 'zod';
+
+import { bulkDownloadOptionsField, bulkDownloadMetadataField } from 'js/components/bulkDownload/bulkDownloadFormFields';
+import useBulkDownloadToasts from 'js/components/bulkDownload/toastHooks';
+import { ALL_BULK_DOWNLOAD_OPTIONS } from 'js/components/bulkDownload/constants';
+import { BulkDownloadDataset, useBulkDownloadStore } from 'js/stores/useBulkDownloadStore';
+import { useSearchHits } from 'js/hooks/useSearchData';
+import { useProtectedDatasetsForm } from 'js/hooks/useProtectedDatasets';
+import { createDownloadUrl } from 'js/helpers/functions';
+import { checkAndDownloadFile, postAndDownloadFile } from 'js/helpers/download';
+import { trackEvent } from 'js/helpers/trackers';
+import { getIDsQuery } from 'js/helpers/queries';
+
+const schema = z
+  .object({
+    ...bulkDownloadOptionsField,
+    ...bulkDownloadMetadataField,
+  })
+  .partial()
+  .required({ bulkDownloadOptions: true });
+
+export interface BulkDownloadFormTypes {
+  bulkDownloadOptions: string[];
+  bulkDownloadMetadata: boolean;
+}
+
+function useBulkDownloadForm() {
+  const {
+    handleSubmit,
+    control,
+    reset,
+    setValue,
+    formState: { errors },
+    trigger,
+  } = useForm<BulkDownloadFormTypes>({
+    defaultValues: {
+      bulkDownloadOptions: [],
+      bulkDownloadMetadata: false,
+    },
+    mode: 'onChange',
+    resolver: zodResolver(schema),
+  });
+
+  return {
+    handleSubmit,
+    setValue,
+    control,
+    errors,
+    reset,
+    trigger,
+  };
+}
+
+interface BulkDownloadDialogProps {
+  deselectRows?: (uuids: string[]) => void;
+}
+function useBulkDownloadDialog({ deselectRows }: BulkDownloadDialogProps) {
```

Comment:
Perhaps this could work to avoid the empty object prop required to use this without providing a `deselectRows` callback?

```suggestion
function useBulkDownloadDialog(deselectRows?: (uuids: string[]) => void;) {
```

Alternatively, if you foresee potentially adding other props to this hook, the empty object could be a default value instead:

```suggestion
interface BulkDownloadDialogProps {
  deselectRows?: (uuids: string[]) => void;
}
function useBulkDownloadDialog(props: BulkDownloadDialogProps = {}) {
```

Reply:
Good idea!